src/blocks/image-comparison-item/components/Image.js

```diff
@@ -2,33 +2,80 @@
  * External dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Button, Flex, FlexItem } from '@wordpress/components';
 import { MediaUpload, MediaUploadCheck } from '@wordpress/block-editor';
+import { Button, Flex, FlexItem } from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Checks if WP generated the specified image size.
+ *
+ * @param {Object} image the image to pick from
+ * @param {string} size  the size to locate
+ *
+ * @return {boolean} Whether or not it has default image size.
+ */
+function hasSize(image, size) {
+  return (
+    'url' in (image?.sizes?.[size] ?? {}) ||
+    'source_url' in (image?.media_details?.sizes?.[size] ?? {})
+  );
+}
+
+/**
+ * Retrieve the URI for an image.
+ *
+ * @param {Object} image the image to pick from
+ * @param {String} size  the size to locate
+ *
+ * @return {String} the located image size URI.
+ */
+function imageUrl(image, size) {
+  return image?.sizes?.[size]?.url || image?.media_details?.sizes?.[size]?.source_url || image?.url;
+}
 
 /**
  * Renders image in various states
  *
  * @param {object}   props                 Component props
  * @param {object}   props.blockProps      Block generated props
- * @param {number}   props.id              Selected image id
- * @param {string}   props.url             Selected image url
+ * @param {object}   props.attributes      Block attributes
  * @param {string}   props.alternativeText Alternative image text
  * @param {function} props.setAttributes   Update block attributes
  */
-const Image = ({ blockProps, id, url, alternativeText, setAttributes }) => {
-  if (id && url) {
+const Image = ({ blockProps, attributes, alternativeText, setAttributes }) => {
+  const media = useSelect(
+    (select) => attributes?.id && select(coreStore).getMedia(attributes.id, { context: 'view' }),
+    [attributes],
```

Comment:
This dependency could be updated to be more specific, since we only care about the `id`:
```suggestion
    [attributes.id],
```

Reply:
changed in 05cd70f